src/files.js

```diff
@@ -0,0 +1,22 @@
+const fs = require('node:fs');
+const path = require('node:path');
+
+const readfileStream = (res, filePath, code, message) => {
+  const fileStream = fs.createReadStream(
+    path.join(__dirname, `../public/${filePath}`),
+  );
+
+  res.statusMessage = message;
+  res.statusCode = code;
+
+  fileStream.pipe(res);
+
+  fileStream.on('error', () => {
+    res.statusCode = 500;
+    res.end('Server Error');
+  });
```

Comment:
Consider adding `res.end()` after setting the status code and message in the error handler to ensure the response is properly terminated in case of an error.